### PR TITLE
Carbon 213 - Add Category to table and columns

### PR DESCRIPTION
**Pull request by ayeswarya**
Implements a feature in https://github.com/yahoo/elide/issues/1178 

## Description
Add Category to Table, Column and Metric function
## Motivation and Context
For NAvi integration

## How Has This Been Tested?
Tested in Spring boot example

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by moizarafat** [comment]
LGTM

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/MetricFunction.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
 
     private String description;
 
+    private String category;
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/MetricFunction.java:33]: WHy do MetricFunctions need categories - when the associated metric already has one?

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/AggregationDataStoreIntegrationTest.java [hunk 2]
```diff
@@ -749,6 +748,24 @@ public void metaDataTest() {
                 .body("data.attributes.expression",  equalTo("player.name"))
                 .body("data.relationships.table.data.id", equalTo("playerStats"));
 
+        given()
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/AggregationDataStoreIntegrationTest.java:751]: This looks like a separate test.  Let's not complicate existing tests by joining them together.  Also, there are already metadata tests - let's see how this new test relates to those.

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/MetricFunction.java (hunk 1 above)
Comment (ayeswarya) [reply elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/MetricFunction.java:33]: https://github.com/yahoo/elide/issues/1005#issuecomment-541196843 i saw this in the RFC

**Pull request merged**